docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,474 @@
+      version:
+        min: TLSv1_2 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+        max: TLSv1_3 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+      allowRenegotiation: false
+      verification:
+        skip: true # if this is true then subjectAltNames don't take effect
+        subjectAltNames: # if subjectAltNames is not defined then take domain or ips
+          - example.com
+          - "spiffe://example.local/ns/local"
+      caCert: 
+        inline: 123
+      clientCert:
+        secret: 123
+      clientKey:
+        secret: 123
+status: 
+  vips:
+  - ip: 242.0.0.1
+    type: Kuma
+    hostname: example.ext.svc.local
+```
+* **spec**:
+  * **match**: defines a list of internalVIPs/domains/CIDR/IPs that should be routed through the sidecar
+    * **type**: type of the match, one of `InternalVIP`, `Domain`, `CIDR` and `IP` are available
+      * `InternalVIP`: allocates a VIP for a domain provided in the `value` field
+      * `Domain`: handles traffic to the specified domain
+      * `CIDR`: handles traffic to specified addresses range
+      * `IP` handles the traffic to specified IP
+    * **value**: depends of the type can be an existing domain, new domain name, CIDR or IP
+    * **port**: defines a port to which a user does requests
+    * **protocol**: defines a protocol of the communication. Possible values:
+      * `tls`: should be used when TLS traffic is originated by the client application
+      * `tcp`: WARNING: shouldn't be used when match has only domains. On the TCP level we are not able to disinguish domain, in this case it is going to hijack whole traffic on this port.
+      * `grpc`
+      * `http`
+      * `http2`
+  * **destination**: defines where matched requests should be routed
+    * **type**: defines what kind of destination is it, one of `Regular`, `Passthrough`, or `Extension`, (Default: `Passthrough`)
+      * `Regular`: allows creating a set of destination endpoints and `TLS` configuration, when defined section `extension` is not available.
+      * `Passthrough`: traffic just passes a proxy without any modifications to the original destination, when defined sections `endpoints`, `tls` and `extension` are not available.
+      * `Extension`: allows specifying a custom plugin for example, user can create a plugin which support AWS Lambda, when defined sections `endpoints` and `tls` are not available.
+    * **extension**: struct for a plugin configuration
+      * **type**: defines what kind of plugin to use, it's a string type so any new plugins should works.
+      * **config**: json map that is mapped to configuration provided in the type.
+    * **endpoints**: defines a list of endpoints
+      * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+      * **port**: defines a port of a destination.
+    * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
```

Comment:
we need to remember about this https://github.com/kumahq/kuma/issues/4758 - and probably mention in the docs that by default it uses system bundle

Reply:
Using system bundle should be default when tls is enabled unless explicitly opted-out